### PR TITLE
Add debug logging.

### PR DESCRIPTION
**Pull request by vkryachko**

### CODE REVIEW

File: transport/transport-runtime/src/main/java/com/google/android/datatransport/runtime/scheduling/jobscheduling/JobInfoScheduler.java [hunk 1]
```diff
@@ -89,14 +92,20 @@ public void schedule(TransportContext transportContext, int attemptNumber) {
...
     int jobId = getJobId(transportContext);
     // Check if there exists a job scheduled for this backend name.
-    if (isJobServiceOn(jobScheduler, jobId, attemptNumber)) return;
+    if (isJobServiceOn(jobScheduler, jobId, attemptNumber)) {
+      Logging.d(
+          LOG_TAG, "Upload for context %s is already scheduled. Returning...", transportContext);
+      return;
+    }
+
+    long nextCallTime = eventStore.getNextCallTime(transportContext);
```
Comment (VinayGuthal) [inline transport/transport-runtime/src/main/java/com/google/android/datatransport/runtime/scheduling/jobscheduling/JobInfoScheduler.java]: why is this needed?

File: transport/transport-runtime/src/main/java/com/google/android/datatransport/runtime/logging/Logging.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  public static void d(String tag, String message, Object arg1, Object arg2) {
+    Log.d(getTag(tag), String.format(message, arg1, arg2));
+  }
+
+  public static void d(String tag, String message, Object... args) {
+    Log.d(getTag(tag), String.format(message, args));
+  }
+
+  public static void i(String tag, String message) {
+    Log.d(getTag(tag), message);
```
Comment (VinayGuthal) [inline transport/transport-runtime/src/main/java/com/google/android/datatransport/runtime/logging/Logging.java]: Log.i maybe?

File: transport/transport-runtime/src/main/java/com/google/android/datatransport/runtime/scheduling/jobscheduling/JobInfoScheduler.java (hunk 1 above)
Comment (vkryachko) [reply transport/transport-runtime/src/main/java/com/google/android/datatransport/runtime/scheduling/jobscheduling/JobInfoScheduler.java]: It's needed to be able to log it below

File: transport/transport-backend-cct/src/main/java/com/google/android/datatransport/cct/CctTransportBackend.java [hunk 3]
```diff
@@ -54,13 +55,11 @@
...
 import java.util.Map;
 import java.util.TimeZone;
-import java.util.logging.Level;
-import java.util.logging.Logger;
 import java.util.zip.GZIPOutputStream;
 
 final class CctTransportBackend implements TransportBackend {
 
-  private static final Logger LOGGER = Logger.getLogger(CctTransportBackend.class.getName());
+  private static final String LOG_TAG = "CctTransportBackend";
```
Comment (rlazo) [inline transport/transport-backend-cct/src/main/java/com/google/android/datatransport/cct/CctTransportBackend.java]: Any reason for not preferring the class.getname()?

File: transport/transport-backend-cct/src/main/java/com/google/android/datatransport/cct/CctTransportBackend.java [hunk 4]
```diff
@@ -243,7 +243,8 @@ private HttpResponse doSend(HttpRequest request) throws IOException {
       }
       channel.write(ByteBuffer.wrap(output.toByteArray()));
       int responseCode = connection.getResponseCode();
-      LOGGER.info("Status Code: " + responseCode);
+      Logging.i(LOG_TAG, "Status Code: " + responseCode);
```
Comment (rlazo) [inline transport/transport-backend-cct/src/main/java/com/google/android/datatransport/cct/CctTransportBackend.java]: Use %s instead of +

File: transport/transport-backend-cct/src/main/java/com/google/android/datatransport/cct/CctTransportBackend.java (hunk 3 above)
Comment (vkryachko) [reply transport/transport-backend-cct/src/main/java/com/google/android/datatransport/cct/CctTransportBackend.java]: yes, `class.getName()` will return the name of the class after proguarding, so it could end up being "a", so to avoid it I explicitly preserve the class name as a constant string

Comment (rlazo) [reply transport/transport-backend-cct/src/main/java/com/google/android/datatransport/cct/CctTransportBackend.java]: Makes sense

File: transport/transport-backend-cct/src/main/java/com/google/android/datatransport/cct/CctTransportBackend.java (hunk 4 above)
Comment (vkryachko) [reply transport/transport-backend-cct/src/main/java/com/google/android/datatransport/cct/CctTransportBackend.java]: Yeah, so I decided not to add an overload that takes format args and did not want to use String.format() directly here as this case is too simple. As we add more uses of info logging we can revisit this.
